tests/js/server/aql/aql-index-choice.js

```diff
@@ -0,0 +1,513 @@
+/*jshint globalstrict:false, strict:false, maxlen: 7000 */
+/*global assertEqual, assertTrue, assertFalse, AQL_EXPLAIN */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief tests for index selection
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2012 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+/// @author Copyright 2012, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const db = require("internal").db;
+const deriveTestSuite = require('@arangodb/test-helper').deriveTestSuite;
+const cn = 'UnitTestsCollection';
+
+let setupCollection = function(cn, n) {
+  db._drop(cn);
+  let c = db._create(cn);
+  let docs = [];
+  for (let i = 0; i < n; ++i) {
+    docs.push({ _key: "test" + i, uid: i, pid: (i % 100), dt: Date.now() - ((i * 10) % 10000) });
+    if (docs.length === 5000) {
+      c.insert(docs);
+      docs = [];
+    }
+  }
+  if (docs.length) {
+    c.insert(docs);
+  }
+};
+
+
+function BaseTestConfig () {
+  let resetIndexes = function(cn) {
+    db[cn].indexes().filter((idx) => idx.type !== 'primary').forEach((idx) => {
+      db[cn].dropIndex(idx);
+    });
+  };
+
+  return {
+    tearDownAll : function() {
+      db._drop(cn);
+    },
+
+    setUp: function() {
+      resetIndexes(cn);
+    },
+    
+    tearDown: function() {
+      resetIndexes(cn);
+    },
+    
+    testSingleIndexSingleFilter: function() {
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid"] });
+      
+      [
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid > 1234 RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 RETURN doc`
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(0, nodes.filter((n) => n.type === 'SortNode').length);
```

Comment:
Why the check that no SortNode is found?

Reply:
Removed this superfluous check :+1: